platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -93,6 +84,47 @@ - (void)getCoordinates:(CLLocationCoordinate2D *)coords range:(NSRange)range
     }
 }
 
+- (void)setCoordinates:(CLLocationCoordinate2D *)coords range:(NSRange)range
+{
+    if ([self isMemberOfClass:[MGLMultiPoint class]])
+    {
+        [NSException raise:@"Unsupported method" format:@"The coordinates of a member of MGLMultiPoint are read-only."];
+    }
+
+    if (range.length == 0)
+    {
+        [[NSException exceptionWithName:NSRangeException
+                                 reason:[NSString stringWithFormat:@"Empty coordinate range %@", NSStringFromRange(range)]
+                               userInfo:nil] raise];
+    }
+    else if (range.location > _count)
+    {
+        [[NSException exceptionWithName:NSRangeException
+                                 reason:[NSString stringWithFormat:
+                                            @"Invalid range %@ for existing coordinate count %zu",
+                                            NSStringFromRange(range), _count]
+                               userInfo:nil] raise];
+    }
+
+    [self willChangeValueForKey:@"coordinates"];
+    if (NSMaxRange(range) <= _count)
+    {
+        // replacing existing coordinate(s)
+        memcpy(_coordinates + range.location, coords, range.length * sizeof(CLLocationCoordinate2D));
```

Comment:
[`memcpy()` documentation](http://www.cplusplus.com/reference/cstring/memcpy/) states that the destination and source of a copy mustn’t overlap, or Bad Things™ can happen. So we should avoid calling `memcpy()` in the case that either `coords` or `coords + coords.length` falls between `_coordinates + range.location` and `_coordinates + NSMaxRange(range)`. Since this is a bit of an edge case – you’d have to pass in the `coordinates` property as `coords` – it’s fine to punt on that by raising an exception, but we should ticket it out as tail work.